app.js

```diff
@@ -9,18 +9,43 @@ function handleError(err) {
   log.error('Processing failed', err);
 }
 
-function app() {
-  return Promise
-    .all([
-      log.info('Starting download and transformation of POMI data'),
-      log.time('Downloading and transforming POMI data took')])
-    .then(downloadFile)
-    .then(removeColumns)
+function startTimer(msg, timerMsg) {
+  return new Promise((resolve, reject) => {
+    try {
+      log.info(msg);
+      log.time(timerMsg);
+      resolve();
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function downloadAndProcessBookingSystem() {
+  const timerMsg = 'Downloading and transforming Booking System data took';
+  return startTimer('Starting download and transformation of Booking System data', timerMsg)
+    .then(downloadFile.booking)
+    .then(removeColumns.booking)
+    .then(getLatestPeriod)
+    .then(removeOldRecords.booking)
+    .then(convertToJson.booking)
+    .then(() => log.timeEnd(timerMsg))
+    .catch(handleError);
+}
+
+function downloadAndProcessRepeatScripts() {
+  const timerMsg = 'Download and transforming Online Repeat Prescriptions data took';
+  return startTimer('Starting download and transform of Repeat Prescription data', timerMsg)
+    .then(downloadFile.scripts)
+    .then(removeColumns.scripts)
     .then(getLatestPeriod)
-    .then(removeOldRecords)
-    .then(convertToJson)
-    .then(() => log.timeEnd('Downloading and transforming POMI data took'))
+    .then(removeOldRecords.scripts)
+    .then(convertToJson.scripts)
+    .then(() => log.timeEnd(timerMsg))
     .catch(handleError);
 }
```

Comment:
It seems there's a lot of duplication between the downloadAndProcess functions, in that the only thing different between them seems to be the file downloaded, and the name of the file created.

As it stands adding another file would mean another download function and more functions in each step.

An approach where you have a downloadAndProcess function that accepted a settings object with url, name, columns, etc would reduce the duplication and make adding the next one quicker.